bridges/snowbridge/primitives/router/src/inbound/v2.rs

```diff
@@ -2,454 +2,172 @@
+use xcm::{
+	prelude::{Junction::AccountKey20, *},
+	MAX_XCM_DECODE_DEPTH,
+};
 
-const MINIMUM_DEPOSIT: u128 = 1;
+const LOG_TARGET: &str = "snowbridge-router-primitives";
 
 /// Messages from Ethereum are versioned. This is because in future,
 /// we may want to evolve the protocol so that the ethereum side sends XCM messages directly.
 /// Instead having BridgeHub transcode the messages into XCM.
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
 pub enum VersionedMessage {
-	V1(MessageV1),
+	V2(Message),
 }
 
-/// For V1, the ethereum side sends messages which are transcoded into XCM. These messages are
+/// For V2, the ethereum side sends messages which are transcoded into XCM. These messages are
 /// self-contained, in that they can be transcoded using only information in the message.
-#[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub struct MessageV1 {
-	/// EIP-155 chain id of the origin Ethereum network
-	pub chain_id: u64,
-	/// The command originating from the Gateway contract
-	pub command: Command,
-}
-
-#[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub enum Command {
-	/// Register a wrapped token on the AssetHub `ForeignAssets` pallet
-	RegisterToken {
-		/// The address of the ERC20 token to be bridged over to AssetHub
-		token: H160,
-		/// XCM execution fee on AssetHub
-		fee: u128,
-	},
-	/// Send Ethereum token to AssetHub or another parachain
-	SendToken {
-		/// The address of the ERC20 token to be bridged over to AssetHub
-		token: H160,
-		/// The destination for the transfer
-		destination: Destination,
-		/// Amount to transfer
-		amount: u128,
-		/// XCM execution fee on AssetHub
-		fee: u128,
-	},
-	/// Send Polkadot token back to the original parachain
-	SendNativeToken {
-		/// The Id of the token
-		token_id: TokenId,
-		/// The destination for the transfer
-		destination: Destination,
-		/// Amount to transfer
-		amount: u128,
-		/// XCM execution fee on AssetHub
-		fee: u128,
-	},
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct Message {
+	/// The origin address
+	pub origin: H160,
+	/// The assets
+	pub assets: Vec<InboundAsset>,
+	// The command originating from the Gateway contract
+	pub xcm: Vec<u8>,
+	// The claimer in the case that funds get trapped.
+	pub claimer: Option<Vec<u8>>,
 }
 
-/// Destination for bridged tokens
-#[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub enum Destination {
-	/// The funds will be deposited into account `id` on AssetHub
-	AccountId32 { id: [u8; 32] },
-	/// The funds will deposited into the sovereign account of destination parachain `para_id` on
-	/// AssetHub, Account `id` on the destination parachain will receive the funds via a
-	/// reserve-backed transfer. See <https://github.com/paritytech/xcm-format#depositreserveasset>
-	ForeignAccountId32 {
-		para_id: u32,
-		id: [u8; 32],
-		/// XCM execution fee on final destination
-		fee: u128,
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum InboundAsset {
```

Comment:
Prefer not to prefix with `Inbound`.

```suggestion
pub enum Asset {
```

Reply:
https://github.com/claravanstaden/polkadot-sdk/pull/4/files#diff-011a2c280539a541db601aa2fd4e5f30960525c4bb35588c3de78effb01720fcR43